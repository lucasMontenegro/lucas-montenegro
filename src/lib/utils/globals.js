import g from "global"
// renaming "global" to "globals"
// declaring a variable named "global" may cause trouble
export default g