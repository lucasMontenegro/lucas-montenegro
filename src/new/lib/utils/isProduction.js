import global from "global"
export default function isProduction () {
  return (
    global.process ? global.process.env.NODE_ENV === `production` :
    process.env.NODE_ENV === `production`
  )
}