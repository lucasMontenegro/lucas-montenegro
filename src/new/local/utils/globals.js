// renaming "global" to "globals"
// declaring a variable named "global" may cause troubles
exports.default = require("global")