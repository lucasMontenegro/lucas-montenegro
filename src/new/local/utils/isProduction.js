const { default: globals } = require("new/local/utils/globals")
exports.default = function isProduction () {
  return (
    globals.process ? globals.process.env.NODE_ENV === `production` :
    process.env.NODE_ENV === `production`
  )
}