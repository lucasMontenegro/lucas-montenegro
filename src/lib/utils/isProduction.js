import globals from "lib/utils/globals"
export default function isProduction () {
  return (
    globals.process ? globals.process.env.NODE_ENV === `production` :
    process.env.NODE_ENV === `production`
    // WebPack will replace this last line with a boolean
  )
}