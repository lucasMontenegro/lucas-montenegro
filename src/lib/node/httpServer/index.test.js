import globals from "lib/utils/globals"
import express from "express"
import isProduction from "lib/utils/isProduction"
jest.mock(`lib/utils/globals`, () => ({ __esModule: true, default: {} }))
jest.mock(`express`, () => ({ __esModule: true, default: jest.fn() }))
express.static = str => `express.static('${str}')`
jest.mock(`lib/utils/isProduction`, () => ({ __esModule: true, default: jest.fn() }))
describe(`./index.js`, () => {
  let httpServer
  beforeAll(() => {
    httpServer = require("./index.js").default
  })
  it(`should use the right versions of its dependencies`, () => {
    expect(jestUtils.getDependencies([`express`])).toMatchSnapshot()
  })
  const cases = [
    [
      1234,
      { name: `api` },
      `Node cluster worker 'globals.process.pid': listening on port 1234`,
      `Node dev server: listening on port 1234`,
    ],
    [
      undefined,
      { name: `api` },
      `Node cluster worker 'globals.process.pid': listening on port 5000`,
      `Node dev server: listening on port 5000`,
    ],
    [
      1234,
      undefined,
      `Node cluster worker 'globals.process.pid': listening on port 1234`,
      `Node dev server: listening on port 1234`,
    ],
    [
      undefined,
      undefined,
      `Node cluster worker 'globals.process.pid': listening on port 5000`,
      `Node dev server: listening on port 5000`,
    ],
  ]
  describe.each(cases)(`httpServer (port %j, api %j)`, (port, api, prodMsg, devMsg) => {
    const appFn = jest.fn()
    const app = {
      use: jest.fn((...args) => appFn(`app.use`, args)),
      get: jest.fn((...args) => appFn(`app.get`, args)),
      listen: jest.fn((...args) => appFn(`app.listen`, args)),
    }
    beforeAll(() => {
      globals.process = { env: { BUILD_PATH: `/build-path`, PORT: port } }
      express.mockReturnValueOnce(app)
      httpServer(api)
    })
    it(`should set up an ExpressJS app`, () => {
      expect(appFn.mock.calls).toMatchSnapshot()
      appFn.mockClear()
    })
    describe(`get * request handler`, () => {
      it(`should send index.html`, () => {
        const handler = app.get.mock.calls[0][1]
        const res = { sendFile: str => `res.sendFile('${str}')` }
        expect(handler({}, res)).toBe(`res.sendFile('/build-path/index.html')`)
      })
    })
    describe(`http listen handler (isProduction => true)`, () => {
      it(`should log a message`, () => {
        const handler = app.listen.mock.calls[0][1]
        globals.process = { pid: `'globals.process.pid'` }
        const log = jest.fn()
        globals.console = { log }
        isProduction.mockReturnValueOnce(true)
        handler()
        expect(log.mock.calls[0][0]).toBe(prodMsg)
      })
    })
    describe(`http listen handler (isProduction => false)`, () => {
      it(`should log a message`, () => {
        const handler = app.listen.mock.calls[0][1]
        const log = jest.fn()
        globals.console = { log }
        isProduction.mockReturnValueOnce(false)
        handler()
        expect(log.mock.calls[0][0]).toBe(devMsg)
      })
    })
  })
})