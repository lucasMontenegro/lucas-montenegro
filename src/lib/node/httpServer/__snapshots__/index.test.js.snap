// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`./index.js httpServer (port 1234, api {"name":"api"}) should set up an ExpressJS app 1`] = `
Array [
  Array [
    "app.use",
    Array [
      "express.static('/build-path')",
    ],
  ],
  Array [
    "app.use",
    Array [
      "/api",
      Object {
        "name": "api",
      },
    ],
  ],
  Array [
    "app.get",
    Array [
      "*",
      [Function],
    ],
  ],
  Array [
    "app.listen",
    Array [
      1234,
      [Function],
    ],
  ],
]
`;

exports[`./index.js httpServer (port 1234, api undefined) should set up an ExpressJS app 1`] = `
Array [
  Array [
    "app.use",
    Array [
      "express.static('/build-path')",
    ],
  ],
  Array [
    "app.get",
    Array [
      "*",
      [Function],
    ],
  ],
  Array [
    "app.listen",
    Array [
      1234,
      [Function],
    ],
  ],
]
`;

exports[`./index.js httpServer (port undefined, api {"name":"api"}) should set up an ExpressJS app 1`] = `
Array [
  Array [
    "app.use",
    Array [
      "express.static('/build-path')",
    ],
  ],
  Array [
    "app.use",
    Array [
      "/api",
      Object {
        "name": "api",
      },
    ],
  ],
  Array [
    "app.get",
    Array [
      "*",
      [Function],
    ],
  ],
  Array [
    "app.listen",
    Array [
      5000,
      [Function],
    ],
  ],
]
`;

exports[`./index.js httpServer (port undefined, api undefined) should set up an ExpressJS app 1`] = `
Array [
  Array [
    "app.use",
    Array [
      "express.static('/build-path')",
    ],
  ],
  Array [
    "app.get",
    Array [
      "*",
      [Function],
    ],
  ],
  Array [
    "app.listen",
    Array [
      5000,
      [Function],
    ],
  ],
]
`;

exports[`./index.js should use the right versions of its dependencies 1`] = `
Object {
  "express": "^4.17.1",
}
`;
